src/helperFunctions/process.py

```diff
@@ -155,12 +184,22 @@ def new_worker_was_started(new_process: ExceptionSafeProcess, old_process: Excep
 
 def stop_processes(processes: List[Process], timeout: float = 10.0):
     '''
-    Try to stop processes gracefully. If a process does not stop until `timeout` is reached, terminate it.
+    Try to stop processes gracefully in parallel. If a process does not stop until `timeout` is reached, kill it.
 
     :param processes: The list of processes that should be stopped.
     :param timeout: Timeout for joining the process in seconds.
     '''
+    thread_list = []
     for process in processes:
-        process.join(timeout=timeout)
-        if process.is_alive():
-            process.terminate()
+        thread = Thread(target=stop_process, args=(process, timeout))
+        thread.start()
+        thread_list.append(thread)
+    for thread in thread_list:
+        thread.join()
+
+
+def stop_process(process: Process, timeout: float = 10.0):
```

Comment:
Should we maybe reconsider the timeout?
Does it really take that long for a process to exit sucessfully?
Are there processes that do that much cleanup?

Reply:
> We could also make it configurable.

I don't think that is necessary. Why should the user know when to kill the processes?

> `max(cfg.expert_settings.intercom_poll_delay, cfg.expert_settings.block_delay) + 1`

Sounds good. If you like, you could add a comment about why we chose this timeout since it isn't obvious imo.